prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -292,6 +292,13 @@ static void server_send_cm_accept(struct util_wait *wait,
 	err_entry.context = cm_ctx->fid->context;
 	err_entry.err = -ret;
 
+	if (wait_del_ret ) {
+		wait_del_ret = ofi_wait_del_fd(wait, ep->sock);
+		if (wait_del_ret)
+			FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+				"Could not remove fd from wait\n");
+	}
```

Comment:
A similar comment applies to this change as the previous patch.  It looks like there's a bigger race condition occurring.

Reply:
Can you explain what you mean? I have updated the description and the patch a bit - the issue I'm fixing is that cm_ctx was freed, but the fd was kept in polling - next poll round then accessed invalid memory. The part I don't understand yet is where err_entry.err == -104 in our QA testing comes from. Testing does server resets and reboots, but the window for a failing ofi_recv_socket() or ofi_send_socket() is rather small and the issue seems to come up with a higher rate than expected (abpout once a day with a few servers and clients). The wrong error sign is also weird - I have checked all functions fi_eq_write() writes and so far I didn't find any wrong sign conversion. I hope I get a bit more with the additional logging.